tfjs-backend-wasm/src/cc/batch_mat_mul_impl.cc

```diff
@@ -235,14 +235,17 @@ void slow_batch_matmul(const size_t a_id, const size_t* a_shape_ptr,
               float sum = 0.0;
 
               for (size_t k = k0; k < k_block; ++k) {
-                const size_t batch_offset_a =
-                    std::min(b, a_shape[0] - 1) * a_batch;
-                const size_t batch_offset_b =
-                    std::min(b, b_shape[0] - 1) * b_batch;
-                sum +=
-                    a_buf[batch_offset_a + i * a_outer_step +
-                          k * a_inner_step] *
-                    b_buf[k * b_inner_step + j * b_outer_step + batch_offset_b];
+                size_t batch_index_a = b;
+                size_t batch_index_b = b;
+                if (b >= a_shape[0]) {
+                  batch_index_a = b % a_shape[0];
+                } else if (b >= b_shape[0]) {
+                  batch_index_b = b % b_shape[0];
```

Comment:
See my comment in the CPU implementation for why this should work.
```suggestion
                size_t batch_index_a = b % a_shape[0];
                size_t batch_index_b = b % b_shape[0];
```

Reply:
Done.